diesel/src/pg/types/ranges.rs

```diff
@@ -0,0 +1,255 @@
+            } else {
+                Bound::Excluded(value)
+            };
+        }
+
+        Ok((lower_bound, upper_bound))
+    }
+}
+
+impl<ST, T> ToSql<Range<ST>, Pg> for (Bound<T>, Bound<T>)
+where
+    Pg: HasSqlType<ST> + HasSqlType<Range<ST>>,
+    T: ToSql<ST, Pg>,
+{
+    fn to_sql<W: Write>(
+        &self,
+        out: &mut ToSqlOutput<W, Pg>,
+    ) -> Result<IsNull, Box<Error + Send + Sync>> {
+        let mut flags = match self.0 {
+            Bound::Included(_) => RANGE_LB_INC,
+            Bound::Excluded(_) => RangeFlags::empty(),
+            Bound::Unbounded => RANGE_LB_INF,
+        };
+
+        flags |= match self.1 {
+            Bound::Included(_) => RANGE_UB_INC,
+            Bound::Excluded(_) => RangeFlags::empty(),
+            Bound::Unbounded => RANGE_UB_INF,
+        };
+
+        out.write_u8(flags.bits())?;
+
+        match self.0 {
+            Bound::Included(ref value) | Bound::Excluded(ref value) => {
+                let mut buffer = out.with_buffer(Vec::new());
+
+                value.to_sql(&mut buffer)?;
+                out.write_u32::<NetworkEndian>(buffer.len() as u32)?;
+                out.write_all(&buffer)?;
+            }
+            Bound::Unbounded => {}
+        }
+
+        match self.1 {
+            Bound::Included(ref value) | Bound::Excluded(ref value) => {
+                let mut buffer = out.with_buffer(Vec::new());
+
+                value.to_sql(&mut buffer)?;
+                out.write_u32::<NetworkEndian>(buffer.len() as u32)?;
+                out.write_all(&buffer)?;
+            }
+            Bound::Unbounded => {}
+        }
+
+        Ok(IsNull::No)
+    }
+}
+
+impl<T> HasSqlType<Range<T>> for Debug
+where
+    Debug: HasSqlType<T>,
+{
+    fn metadata(_: &()) {}
+}
+
+impl HasSqlType<Range<Int4>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3904,
+            array_oid: 3905,
+        }
+    }
+}
+
+impl HasSqlType<Range<Numeric>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3906,
+            array_oid: 3907,
+        }
+    }
+}
+
+impl HasSqlType<Range<Timestamp>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3908,
+            array_oid: 3909,
+        }
+    }
+}
+
+impl HasSqlType<Range<Timestamptz>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3910,
+            array_oid: 3911,
+        }
+    }
+}
+
+impl HasSqlType<Range<Date>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3912,
+            array_oid: 3913,
+        }
+    }
+}
+
+impl HasSqlType<Range<Int8>> for Pg {
+    fn metadata(_: &PgMetadataLookup) -> PgTypeMetadata {
+        PgTypeMetadata {
+            oid: 3926,
+            array_oid: 3927,
+        }
+    }
+}
```

Comment:
`primitive_impls` but it doesn't like generic stuff and I think it also implements other things.